### PR TITLE
Add catalog tests

### LINKED ISSUE #24: Fix TestCatalog.
This should now be TestHSQLDBCatalog. I haven't looked at updating it to match the new APIs (if any).

https://github.com/cmu-db/benchbase/blob/main/src/test/java/com/oltpbenchmark/catalog/TestCatalog.java

### PR DESCRIPTION
**Pull request by jkosh44**
Fixes #24

### CODE REVIEW

**Comment by jkosh44** [comment]
Just trying to get my Java fix...

File: src/main/java/com/oltpbenchmark/util/SQLUtil.java [hunk 1]
```diff
@@ -420,7 +440,7 @@ public static AbstractCatalog getCatalog(BenchmarkModule benchmarkModule, Databa
      * @return
      * @throws SQLException
```
Comment (lmwnshn) [inline src/main/java/com/oltpbenchmark/util/SQLUtil.java:441]: Outdated line

File: src/test/java/com/oltpbenchmark/catalog/TestHSQLDBCatalog.java [hunk 2]
```diff
@@ -0,0 +1,148 @@
+/******************************************************************************
+ *  Copyright 2015 by OLTPBenchmark Project                                   *
```
Comment (lmwnshn) [inline src/test/java/com/oltpbenchmark/catalog/TestHSQLDBCatalog.java:2]:
```suggestion
 *  Copyright 2021 by OLTPBenchmark Project                                   *
```

File: src/test/java/com/oltpbenchmark/catalog/TestHSQLDBCatalog.java [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+        // Make sure that the key and values in this map are not
+        // equal unless we ignore their case
+        Map<String, String> origTableNames = this.catalog.getOriginalTableNames();
+        assertNotNull(origTableNames);
+        assertFalse(origTableNames.isEmpty());
+
+        for (Entry<String, String> e : origTableNames.entrySet()) {
+            assertFalse(e.toString(), e.getKey().equals(e.getValue()));
+            assertTrue(e.toString(), e.getKey().equalsIgnoreCase(e.getValue()));
+        } // FOR
```
Comment (lmwnshn) [inline src/test/java/com/oltpbenchmark/catalog/TestHSQLDBCatalog.java:57]:
```suggestion
        }
```

We removed this style recently

File: src/test/java/com/oltpbenchmark/catalog/TestHSQLDBCatalog.java [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+            String contents = new String(stream.readAllBytes());
+            assertFalse(contents.isEmpty());
+            int offset = 0;
+            int num_tables = 0;
+            while (offset < contents.length()) {
+                offset = contents.indexOf("CREATE TABLE", offset);
+                if (offset == -1) break;
+                num_tables++;
+                offset++;
+            } // FOR
```
Comment (lmwnshn) [inline src/test/java/com/oltpbenchmark/catalog/TestHSQLDBCatalog.java:77]:
```suggestion
            }
```

File: src/test/java/com/oltpbenchmark/catalog/TestHSQLDBCatalog.java [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+
+            // Make sure that CatalogUtil returns the same number of tables
+            assertEquals(num_tables, this.catalog.getTables().size());
+
+            // Make sure that Map names match the Table names
+            for (String table_name : this.catalog.getTables().stream().map(AbstractCatalogObject::getName).toList()) {
+                Table catalog_tbl = this.catalog.getTable(table_name);
+                assertNotNull(catalog_tbl);
+                assertEquals(table_name, catalog_tbl.getName());
+            } // FOR
```
Comment (lmwnshn) [inline src/test/java/com/oltpbenchmark/catalog/TestHSQLDBCatalog.java:88]:
```suggestion
            }
```

File: src/test/java/com/oltpbenchmark/catalog/TestHSQLDBCatalog.java [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+        int found = 0;
+        assertNotNull(catalog_tbl);
+        for (Column catalog_col : catalog_tbl.getColumns()) {
+            assertNotNull(catalog_col);
+            Column fkey_col = catalog_col.getForeignKey();
+            if (fkey_col != null) {
+                assertNotEquals(fkey_col.getTable(), catalog_tbl);
+                found++;
+            }
+        } // FOR
```
Comment (lmwnshn) [inline src/test/java/com/oltpbenchmark/catalog/TestHSQLDBCatalog.java:107]:
```suggestion
        }
```

File: src/test/java/com/oltpbenchmark/catalog/TestHSQLDBCatalog.java [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+
+            for (Index catalog_idx : catalog_tbl.getIndexes()) {
+                assertNotNull(catalog_idx);
+                assertEquals(catalog_tbl, catalog_idx.getTable());
+                assertTrue(catalog_idx.getColumns().size() > 0);
+
+                for (int i = 0; i < catalog_idx.getColumns().size(); i++) {
+                    assertNotNull(catalog_idx.getColumns().get(i).getName());
+                    assertNotNull(catalog_idx.getColumns().get(i).getDir());
+                } // FOR
```
Comment (lmwnshn) [inline src/test/java/com/oltpbenchmark/catalog/TestHSQLDBCatalog.java:126]:
```suggestion
                }
```

File: src/test/java/com/oltpbenchmark/catalog/TestHSQLDBCatalog.java [hunk 8]
```diff
@@ -0,0 +1,148 @@
...
+            } // FOR
```
Comment (lmwnshn) [inline src/test/java/com/oltpbenchmark/catalog/TestHSQLDBCatalog.java:127]:
```suggestion
            }
```

File: src/test/java/com/oltpbenchmark/catalog/TestHSQLDBCatalog.java [hunk 9]
```diff
@@ -0,0 +1,148 @@
...
+        // So we need to check to make sure that our little checker works
+        for (Table catalog_tbl : this.catalog.getTables()) {
+            assertNotNull(catalog_tbl);
+            for (Column catalog_col : catalog_tbl.getColumns()) {
+                assertNotNull(catalog_col);
+                if (catalog_col.getName().contains("_IATTR")) {
+                    boolean actual = SQLUtil.isIntegerType(catalog_col.getType());
+                    assertTrue(catalog_col.getName() + " -> " + catalog_col.getType(), actual);
+                }
+            } // FOR (col)
```
Comment (lmwnshn) [inline src/test/java/com/oltpbenchmark/catalog/TestHSQLDBCatalog.java:145]:
```suggestion
            }
```

File: src/test/java/com/oltpbenchmark/catalog/TestHSQLDBCatalog.java [hunk 10]
```diff
@@ -0,0 +1,148 @@
...
+        } // FOR (table)
```
Comment (lmwnshn) [inline src/test/java/com/oltpbenchmark/catalog/TestHSQLDBCatalog.java:146]:
```suggestion
        }
```

**Review by lmwnshn — APPROVED** [review]
LGTM, minor nits

File: src/main/java/com/oltpbenchmark/util/SQLUtil.java (hunk 1 above)
Comment (jkosh44) [reply src/main/java/com/oltpbenchmark/util/SQLUtil.java]: Removed

**Pull request merged**